ArcGISRuntimeSDKQt_QMLSamples/UtilityNetwork/CreateLoadReport/CreateLoadReport.qml

```diff
@@ -0,0 +1,484 @@
+            utilityNetwork.load();
+        }
+
+        onLoadStatusChanged: {
+            if (utilityNetwork.loadStatus === Enums.LoadStatusLoaded) {
+                utilityAssetType = createUtilityAssetType();
+                utilityTier = createUtilityTier();
+
+                // Create a UtilityElement from the UtilityAssetType to use as the starting location
+                startingLocation = createStartingLocation();
+
+                // Get a default trace configuration from a tier in the network.
+                traceParams.traceConfiguration = createTraceConfiguration();
+
+                // Create a base condition to compare against
+                baseCondition = traceParams.traceConfiguration.traversability.barriers;
+
+                // Create a list of possible phases from a given network attribute
+                phaseCodedValuesList = createCodedValuesPhaseList();
+                sampleStatus = 2;
+            }
+        }
+
+        onTraceStatusChanged: {
+            if (traceStatus === Enums.TaskStatusCompleted) {
+                let customers = 0;
+                let load = 0;
+
+                traceResult.forEach(result => {
+                                        if (result.objectType === "UtilityElementTraceResult")
+                                            customers = result.elements.length;
+
+                                        else if (result.objectType === "UtilityFunctionTraceResult")
+                                            load = result.functionOutputs[0].result;
+                                    });
+
+                setGridText(currentPhase.name, customers, load);
+
+                if (phaseQueue.length > 0) {
+                    currentPhase = phaseQueue.pop();
+                    createReportForPhase(currentPhase);
+                } else {
+                    sampleStatus = 2;
+                    reportHasRun = true;
+                }
+            }
+        }
+
+        onErrorChanged: {
+            sampleStatus = -1;
+        }
+    }
+
+    UtilityCategoryComparison {
+        // Create a comparison to check the existence of service points.
+        id: serviceCategoryComparison
+        comparisonOperator: Enums.UtilityCategoryComparisonOperatorExists
+
+        // Service Category for counting total customers
+        function setServiceCategoryComparison() {
+            let utilityCategories = utilityNetwork.definition.categories;
+            for (let i = 0; i < utilityCategories.length; i++) {
+                if (utilityCategories[i].name === serviceCategoryName) {
+                    serviceCategoryComparison.category = utilityCategories[i];
+                    break;
+                }
+            }
+        }
+    }
+
+    UtilityTraceFunction {
+        id: addLoadAttributeFunction
+        functionType: Enums.UtilityTraceFunctionTypeAdd
+        condition: serviceCategoryComparison
+
+        function setNetworkAttribute() {
```

Comment:
why wrap this property in a setter function? could you just set the property from wherever the function is called?